doc/debug_presubmit.md

```diff
@@ -0,0 +1,40 @@
+# Debug ESP presubmit failures on Jenkins
+
+ESP runs the presubmit unit/integration/e2e tests on Jenkins. The local tests pass
+may not necessarily make ESP presubmit pass as different environment. The most common reasons is
+some dependency incompatibility.
+
+## How to locally reproduce the presubmit failure
+
+#### Getting the presubmit image
+The address of docker image used to run presubmit is is located in [Jenkinsfile](https://github.com/cloudendpoints/esp/blob/master/Jenkinsfile) under var `SLAVE_IMAGE`, like 
+```
+SLAVE_IMAGE = 'gcr.io/endpoints-jenkins/debian-9:0.13'
```

Comment:
Do you know how this image is created? It is 1 year old, and dependencies inside it that are outdated?

Reply:
It is in this [Makefile](https://github.com/cloudendpoints/esp/blob/master/jenkins/slaves/Makefile)